include/rmm/device_scalar.hpp

```diff
@@ -37,18 +37,26 @@ class device_scalar {
  public:
   static_assert(std::is_trivially_copyable<T>::value, "Scalar type must be trivially copyable");
 
-  using value_type      = typename device_uvector<T>::value_type;
-  using reference       = typename device_uvector<T>::reference;
-  using const_reference = typename device_uvector<T>::const_reference;
-  using pointer         = typename device_uvector<T>::pointer;
-  using const_pointer   = typename device_uvector<T>::const_pointer;
+  using value_type =
+    typename device_uvector<T>::value_type;  ///< The type of the elements in the scalar
+  using reference       = typename device_uvector<T>::reference;        ///< Not used
+  using const_reference = typename device_uvector<T>::const_reference;  ///< Not used
+  using pointer =
+    typename device_uvector<T>::pointer;  ///< The type of the pointer returned by data()
+  using const_pointer = typename device_uvector<T>::const_pointer;  ///< The type of the iterator
+                                                                    ///< returned by data() const
 
   RMM_EXEC_CHECK_DISABLE
   ~device_scalar() = default;
 
   RMM_EXEC_CHECK_DISABLE
-  device_scalar(device_scalar&&) noexcept = default;
+  device_scalar(device_scalar&&) noexcept = default;  ///< Default move constructor
 
+  /**
+   * @brief Move assignment operator is deleted as it doesn't allow a stream argument
```

Comment:
It's not deleted, it's explicitly defaulted. For move assignment, we don't need a stream argument.

Again, I don't think this function needs documentation.